### PR TITLE
working on migrating to v1 of channelable.

### PR DESCRIPTION
**Pull request by n3wscott**
## Proposed Changes

- Migrate the shape of NatssChannel to conform to Channaleble ducktype v1.
- Drop v1alpha1 NatssChannel.
- Introduce v1beta1 NatssChannel

Note: no migration support is being added, and no conversion webhooks.

Clusters that wish to update to the new version of eventing-natss will need to delete their channel resources first and then upgrade the controllers.

**Release Note**

```release-note
No migration support has been added for NatssChannel v1alpha1 to v1beta1 conversion. The v1alpha1 NatssChannel is dropped and removed. Clusters that wish to update to the new version of eventing-natss will need to delete their channel resources first and then upgrade the controller.
```

**Docs**

### CODE REVIEW

File: pkg/apis/messaging/v1beta1/natss_channel_defaults.go [hunk 1]
```diff
@@ -14,13 +14,26 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1alpha1
+package v1beta1
 
 import (
 	"context"
```
Comment (mattmoor) [inline pkg/apis/messaging/v1beta1/natss_channel_defaults.go:20]: Format Go code:
```suggestion
	"context"

```

File: pkg/apis/messaging/v1beta1/natss_channel_lifecycle_test.go [hunk 2]
```diff
@@ -14,20 +14,18 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1alpha1
+package v1beta1
 
 import (
```
Comment (mattmoor) [inline pkg/apis/messaging/v1beta1/natss_channel_lifecycle_test.go:19]: Format Go code:
```suggestion
import (
	"testing"

```

File: pkg/apis/messaging/v1beta1/natss_channel_lifecycle_test.go [hunk 3]
```diff
@@ -14,20 +14,18 @@ See the License for the specific language governing permissions and
...
+	eventingduckv1 "knative.dev/eventing/pkg/apis/duck/v1"
 	"testing"
```
Comment (mattmoor) [inline pkg/apis/messaging/v1beta1/natss_channel_lifecycle_test.go:21]: Format Go code:
```suggestion

```

File: pkg/apis/messaging/v1beta1/natss_channel_types_test.go [hunk 4]
```diff
@@ -14,9 +14,10 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1alpha1
+package v1beta1
 
 import (
```
Comment (mattmoor) [inline pkg/apis/messaging/v1beta1/natss_channel_types_test.go:19]: Format Go code:
```suggestion
import (
	"testing"

```

File: pkg/apis/messaging/v1beta1/natss_channel_types_test.go [hunk 5]
```diff
@@ -14,9 +14,10 @@ See the License for the specific language governing permissions and
...
+	eventingduckv1 "knative.dev/eventing/pkg/apis/duck/v1"
 	"testing"
```
Comment (mattmoor) [inline pkg/apis/messaging/v1beta1/natss_channel_types_test.go:21]: Format Go code:
```suggestion

```

File: pkg/apis/messaging/v1beta1/natss_channel_validation.go [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
```
Comment (mattmoor) [inline pkg/apis/messaging/v1beta1/natss_channel_validation.go:21]: Format Go code:
```suggestion
	"fmt"

```

File: pkg/apis/messaging/v1beta1/natss_channel_validation_test.go [hunk 7]
```diff
@@ -14,16 +14,16 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1alpha1
+package v1beta1
 
 import (
 	"context"
```
Comment (mattmoor) [inline pkg/apis/messaging/v1beta1/natss_channel_validation_test.go:20]: Format Go code:
```suggestion
	"context"
	"testing"

```

File: pkg/apis/messaging/v1beta1/natss_channel_validation_test.go [hunk 8]
```diff
@@ -14,16 +14,16 @@ See the License for the specific language governing permissions and
...
+	eventingduckv1 "knative.dev/eventing/pkg/apis/duck/v1"
 	"testing"
```
Comment (mattmoor) [inline pkg/apis/messaging/v1beta1/natss_channel_validation_test.go:22]: Format Go code:
```suggestion

```

File: pkg/dispatcher/dispatcher.go [hunk 9]
```diff
@@ -19,6 +19,7 @@ package dispatcher
 import (
 	"context"
 	"fmt"
+	sdaasdsaddsa "knative.dev/eventing/pkg/apis/messaging/v1beta1"
```
Comment (mattmoor) [inline pkg/dispatcher/dispatcher.go:22]: Format Go code:
```suggestion

```

File: pkg/reconciler/dispatcher/natsschannel.go [hunk 10]
```diff
@@ -19,6 +19,7 @@ package controller
 import (
 	"context"
 	"fmt"
+	eventingduck "knative.dev/eventing/pkg/apis/duck/v1alpha1"
```
Comment (mattmoor) [inline pkg/reconciler/dispatcher/natsschannel.go:22]: Format Go code:
```suggestion

```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

---
Unable to apply comments to the following files: 
 - `pkg/reconciler/controller/resources/service.go`

**Review by antoineco — APPROVED** [review]
/lgtm

**Pull request merged**